src/components/Gauge.tsx

```diff
@@ -0,0 +1,216 @@
+import { Chart } from 'chart.js';
+import { map } from 'lodash';
+import * as React from 'react';
+import { compose } from 'recompose';
+import styled, { withTheme } from 'styled-components';
+
+import asRendition from '../asRendition';
+import { doughnutDefaults, doughnutLabel } from '../chartPlugins';
+import theme from '../theme';
+
+interface ChartEntry {
+	key?: number;
+	value: number;
+	name: string;
+	color: string;
+}
+
+interface GaugeProps {
+	title: string;
+	placeholderColor?: string;
+	data: ChartEntry[];
+}
+
+const { font } = theme;
+
+const StatusGaugeWrapper = styled.div`
+	display: block;
+	color: ${({ theme }) => theme.colors.text.main};
+
+	.gauge__graph {
+		display: block;
+		width: 100%;
+		max-width: 180px;
+		margin: 0px auto 10px;
+	}
+
+	.gauge__legend {
+		list-style: none;
+		padding: 5px;
+		margin: 16px auto 0;
+		background: white;
+		border-radius: 5px;
+		max-width: 240px;
+		border: 1px solid #ececec;
+		overflow: hidden;
+	}
+
+	.gauge__legend-item {
+		display: flex;
+		padding: 3px 0 3px 2px;
+		line-height: 9px;
+		width: 50%;
+		float: left;
+	}
+
+	.gauge__legend-color {
+		display: inline-block;
+		flex: 0 0 auto;
+		width: 10px;
+		height: 10px;
+		margin-right: 4px;
+	}
+
+	.gauge__legend-name {
+		font-size: 11px;
+	}
+`;
+
+class StatusGauge extends React.Component<GaugeProps, {}> {
+	private doughnutChart: Chart | null;
+	private doughnutCtx: CanvasRenderingContext2D | null;
+
+	constructor(props: GaugeProps) {
+		super(props);
+	}
+
+	componentDidMount() {
+		this.generateDoughnut(this.props);
+	}
+
+	componentWillReceiveProps(nextProps: GaugeProps) {
+		this.generateDoughnut(nextProps);
+	}
+
+	componentWillUnmount() {
+		this.destroyChart();
+	}
+
+	generateDoughnut(props: GaugeProps) {
+		if (!this.doughnutCtx) {
+			return false;
+		}
+		// Redraw the chart on props update
+		this.destroyChart();
+
+		const ctx = this.doughnutCtx;
+
+		const { title } = props;
+		const { values, colors, labels, count } = this.normalizeProps(props.data);
+
+		const data = this.getGraphData(values, colors, labels, count);
```

Comment:
Another weird bit of logic here, can you just remove `normalizeProps` and pass `props.data` directly to `getGraphData`. At the moment this is unnecessarily complicated.

Reply:
That would be better